nengo/builder/processes.py

```diff
@@ -161,6 +178,14 @@ def build_synapse(model, synapse, sig_in, sig_out=None):
         sig_out = Signal(
             np.zeros(sig_in.shape), name="%s.%s" % (sig_in.name, synapse))
 
+    shape_in = sig_in.shape if sig_in is not None else (0,)
+    shape_out = sig_out.shape if sig_out is not None else (0,)
+    state_init = synapse.allocate(shape_in, shape_out, model.dt)
+    state = {}
+    for name, value in state_init.items():
+        state[name] = Signal(value)
+        model.sig[synapse][state_prefix + name] = state[name]
```

Comment:
Is there a clean way to avoid copy-pasting this code (here and above)? This would also remove the need for a global `state_prefix` variable?